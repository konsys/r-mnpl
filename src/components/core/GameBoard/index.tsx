import React from "react";

import { useStore } from "effector-react";

import { IFieldModalPosition } from "../../../types/types";
import { GameBoardView } from "./GameBoardView";

export const fieldsActionPosition = (): IFieldModalPosition => {
  return {
    left: 0,
    top: 0,
  };
};

export const GameBoard = () => {
  return <GameBoardView />;
};
