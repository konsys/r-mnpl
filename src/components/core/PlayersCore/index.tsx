import React from "react";
import { RouteComponentProps } from "@reach/router";

interface Props extends RouteComponentProps {}

//TODO unused now, is it needed?
export const Game = (props: Props) => {
  return <>GAME</>;
};
