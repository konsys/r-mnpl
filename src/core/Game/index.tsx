import React, { useState } from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Players } from "../../components/Players/Players";
import { Chat } from "../../components/Chat/Chat";
import { Contract } from "../../components/Contract/Contract";
import { TableHelper } from "../../components/TableHelper/TableHelper";
import { Token } from "../../components/Token/Token";
import { Ticket } from "../../components/Ticket/ticket";
import { Arbitr } from "../../components/Arbitr/Arbitr";
import { M1tv } from "../../components/M1tv/M1tv";
import { TableAction } from "../../components/TableAction/TableAction";
import { Board } from "../../components/Board/Board";
import { Dices } from "../../components/Dices/Dices";
import { GameLoading } from "../../components/GameLoading/GameLoading";
import { dices, diceRandStandart, resetDices } from "./DicesStore";

import { useStore } from "effector-react";
import { calcPosition } from "./TokensStore";

export interface PlayerToken {
  position: number;
  isJailed: number;
}

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const [isGenerators, setIsGenerators] = useState(false);
  const [tableActionVisible, setTableActionVisible] = useState(true);

  const diceStore = useStore(dices);

  const turn = () => {
    resetDices();
    setTableActionVisible(false);
    setIsGenerators(true);
    calcPosition();
    setTimeout(() => diceRandStandart());
    setTimeout(() => {
      setIsGenerators(false);
      setTableActionVisible(true);
    }, 1000);
  };

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <Players />
            <div className="table-body-board">
              <Board />
              <div className="table-body-board-center">
                <M1tv />
                {tableActionVisible && (
                  <TableAction
                    title={"Покупаем?"}
                    text={
                      "Если вы откажетесь от покупки, то поле будет выставлено на общий аукцион."
                    }
                    actions={[{ title: `Бросить кубики`, onClick: turn }]}
                  />
                )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Token id={1} />
              </div>
              {isGenerators && (
                <Dices
                  value1={diceStore.dice1}
                  value2={diceStore.dice2}
                  value3={diceStore.dice3}
                />
              )}
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
    </>
  );
};
