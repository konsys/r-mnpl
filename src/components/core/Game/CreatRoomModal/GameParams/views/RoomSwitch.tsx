import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  text,
  name,
  onChange,
}: {
  text: string;
  name: string;
  onChange: (name: string) => void;
}) {
  const { t } = useTranslation();

  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          color="primary"
          checked={false}
          onChange={() => onChange(name)}
          name={name}
          inputProps={{ "aria-label": name }}
        />
      </Grid>
    </Grid>
  );
}
