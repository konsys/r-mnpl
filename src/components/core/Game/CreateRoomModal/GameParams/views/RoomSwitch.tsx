import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  text,
  name,
  checked,
  onChange,
  disabled,
}: {
  text: string;
  name: string;
  checked: boolean;
  onChange: (name: string) => void;
  disabled: boolean;
}) {
  const { t } = useTranslation();
  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          disabled={disabled}
          color="primary"
          checked={checked}
          onChange={() => onChange(name)}
          name={name}
          inputProps={{ "aria-label": name }}
        />
      </Grid>
    </Grid>
  );
}
