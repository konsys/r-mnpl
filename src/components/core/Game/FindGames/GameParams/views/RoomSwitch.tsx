import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import { IRoomSetup } from "../../CreateGameModal";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  setup,
  text,
  parameterName,
}: {
  setup: IRoomSetup | any;
  text: string;
  parameterName: any;
}) {
  const { t } = useTranslation();

  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          color="primary"
          checked={setup.state[parameterName]}
          onChange={() =>
            setup.setState({
              ...setup.state,
              [parameterName]: !setup.state[parameterName],
            })
          }
          name={parameterName}
          inputProps={{ "aria-label": parameterName }}
        />
      </Grid>
    </Grid>
  );
}
