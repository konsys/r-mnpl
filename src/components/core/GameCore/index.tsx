import React from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Chat } from "../../views/Chat/Chat";
import { Contract } from "../../views/Contract/Contract";
import { TableHelper } from "../../views/TableHelper/TableHelper";
import { Tokens } from "../../views/Tokens/Tokens";
import { Ticket } from "../../views/Ticket/ticket";
import { Arbitr } from "../../views/Arbitr/Arbitr";
import { M1tv } from "../../views/M1tv/M1tv";
import { BoardModal } from "../../views/BoardModal/BoardModal";
import { Dices } from "../../views/Dices/Dices";
import { GameLoading } from "../../views/GameLoading/GameLoading";
import { useStore } from "effector-react";
import { BoardCore } from "../BoardCore/BoardCore";
import { UsersCore } from "../PlayersCore/PlayersCore";
import { actionsStore } from "../../../stores/ActionStore";
import { IncomeMessageType } from "../../../types/BoardTypes";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer } from "react-toastify";

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const actionState = useStore(actionsStore);
  const actionType = actionState.event.action.type;

  const isModalAction =
    actionType === IncomeMessageType.INCOME_AUCTION_MODAL ||
    actionType === IncomeMessageType.INCOME_CAN_BUY_MODAL ||
    actionType === IncomeMessageType.INCOME_ROLL_DICES_MODAL ||
    actionType === IncomeMessageType.INCOME_TAX_PAYING_MODAL ||
    actionType === IncomeMessageType.INCOME_UN_JAIL_MODAL;

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <UsersCore />
            <div className="table-body-board">
              <BoardCore />
              <div className="table-body-board-center">
                <M1tv />
                {actionState && (
                  // TODO check for user credentials
                  // userState.userId === actionState.event.action.userId &&
                  <BoardModal isModal={isModalAction} />
                )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Tokens />
              </div>
              <Dices />
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
      <ToastContainer />
    </>
  );
};
