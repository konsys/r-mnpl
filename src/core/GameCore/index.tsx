import React, { useEffect } from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Chat } from "../../components/Chat/Chat";
import { Contract } from "../../components/Contract/Contract";
import { TableHelper } from "../../components/TableHelper/TableHelper";
import { Token } from "../../components/Token/Token";
import { Ticket } from "../../components/Ticket/ticket";
import { Arbitr } from "../../components/Arbitr/Arbitr";
import { M1tv } from "../../components/M1tv/M1tv";
import { BoardModal } from "../../components/BoardModal/BoardModal";
import { Dices } from "../../components/Dices/Dices";
import { GameLoading } from "../../components/GameLoading/GameLoading";
import { dicesStore } from "../models/DicesStore";
import { useStore } from "effector-react";
import openSocket from "socket.io-client";
import { BoardCore } from "../BoardCore/BoardCore";
import { UsersCore } from "../PlayersCore/PlayersCore";
import { boardMessageHandler } from "./handlers/SocketHandlers";
import nanoid from "nanoid";
import { setBoardIdEvent, resetBoardEvent } from "../models/BoardStore";
import { SocketActions } from "../models/types/ActionsTypes";
import { userStore } from "../models/UserModel";
import { boardActionsStore } from "../models/BoardActionStore";
import { BoardActionType } from "../models/types/BoardTypes";

export const mnplSocket = openSocket("http://localhost:3001");

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const dicesState = useStore(dicesStore);
  const actionState = useStore(boardActionsStore);
  const userState = useStore(userStore);

  useEffect(() => {
    setBoardIdEvent(nanoid(12));
    mnplSocket.on(SocketActions.BOARD_MESSAGE, boardMessageHandler);
    return () => resetBoardEvent();
  }, []);

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <UsersCore />
            <div className="table-body-board">
              <BoardCore />
              <div className="table-body-board-center">
                <M1tv />
                {userState.userId === actionState?.userId &&
                  actionState.action === BoardActionType.SHOW_MODAL && (
                    <BoardModal />
                  )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Token userId={1} />
              </div>
              {actionState?.action === BoardActionType.ROLL_DICES &&
                Array.isArray(dicesState.dices) &&
                dicesState.dices.length === 3 && (
                  <Dices
                    value1={dicesState.dices[0]}
                    value2={dicesState.dices[1]}
                    value3={dicesState.dices[2]}
                  />
                )}
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
    </>
  );
};
