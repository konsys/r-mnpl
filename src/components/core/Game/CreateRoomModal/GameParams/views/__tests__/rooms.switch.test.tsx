import { Button, Switch } from "@material-ui/core";

import React from "react";
import RoomSwitch from "../RoomSwitch";
import { ShallowRendererProps } from "enzyme";
import { createShallow } from "@material-ui/core/test-utils";

describe("Rooms switch test", () => {
  let shallow: ShallowRendererProps | any;

  beforeAll(() => {
    shallow = createShallow() as ShallowRendererProps;
  });

  afterAll(() => {
    shallow.cleanUp();
  });

  it("should render ", () => {
    expect(
      shallow(
        <RoomSwitch
          text={"test"}
          name={"test"}
          checked={true}
          onChange={() => null}
          disabled={false}
        />
      )
    ).toMatchSnapshot();
  });

  it("should render true props", () => {
    const wrap = shallow(
      <RoomSwitch
        text={"testText"}
        name={"testName"}
        checked={false}
        onChange={() => null}
        disabled={false}
      />
    );
    expect(wrap.find(Switch).props("name").name).toStrictEqual("testName");
    expect(wrap.find(Switch).props("name").checked).toStrictEqual(false);
    expect(wrap.find(Switch).props("name").disabled).toStrictEqual(false);
  });

  it("should render false props", () => {
    const wrap = shallow(
      <RoomSwitch
        text={"testText"}
        name={"testName"}
        checked={true}
        onChange={() => null}
        disabled={true}
      />
    );
    expect(wrap.find(Switch).props("name").checked).toStrictEqual(true);
    expect(wrap.find(Switch).props("name").disabled).toStrictEqual(true);
  });
});
