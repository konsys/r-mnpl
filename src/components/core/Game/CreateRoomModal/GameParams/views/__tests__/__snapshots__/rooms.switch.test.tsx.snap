// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms switch test should render  1`] = `ShallowWrapper {}`;
